nodes/rare/rare.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Nokia
+// Licensed under the BSD 3-Clause License.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package rare
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/srl-labs/containerlab/nodes"
+	"github.com/srl-labs/containerlab/types"
+	"github.com/srl-labs/containerlab/utils"
+)
+
+var kindnames = []string{"rare"}
+
+// Register registers the node in the NodeRegistry.
+func Register(r *nodes.NodeRegistry) {
+	r.Register(kindnames, func() nodes.Node {
+		return new(rare)
+	}, nil)
+}
+
+type rare struct {
+	nodes.DefaultNode
+}
+
+func (n *rare) Init(cfg *types.NodeConfig, opts ...nodes.NodeOption) error {
+	// Init DefaultNode
+	n.DefaultNode = *nodes.NewDefaultNode(n)
+
+	n.Cfg = cfg
+	for _, o := range opts {
+		o(n)
+	}
+
+	// make ipv6 disabled on all rare node interfaces unconditionally
+	// as ipv6 will be handled by rare/freertr
+	cfg.Sysctls["net.ipv6.conf.all.disable_ipv6"] = "1"
+
+	n.Cfg.Binds = append(n.Cfg.Binds,
+		fmt.Sprint(filepath.Join(n.Cfg.LabDir, "run"), ":/rtr/run"),
+	)
+
+	return nil
+}
+
+func (n *rare) PreDeploy(_ context.Context, params *nodes.PreDeployParams) error {
+	utils.CreateDirectory(n.Cfg.LabDir, 0777)
+	_, err := n.LoadOrGenerateCertificate(params.Cert, params.TopologyName)
+	if err != nil {
+		return nil
+	}
+
+	return createRAREFiles(n)
+}
+
+func createRAREFiles(node nodes.Node) error {
```

Comment:
can we make this a method, please.
node.Config() is then availabe already as n.Cfg.

Reply:
While double-checking I see SROS is implemented in the same way. Should maybe also be altered seperately.

Reply:
agree, I did it in 1259b67

Reply:
I just implemented `xrd``PR#1144` example provided by @hellt and got inspired with `crpd`(as It happened that i already had a `crpd` node). And yes, you are absolutely right this can be altered separately in a global approach. Because it is also present in :
* [cRPD](https://github.com/srl-labs/containerlab/blob/main/nodes/crpd/crpd.go#L76)

```
❯ grep -r create ./nodes | egrep "nodes.Node"
./nodes/rare/rare.go:func createRAREFiles(node nodes.Node) error {
./nodes/crpd/crpd.go:func createCRPDFiles(node nodes.Node) error {
./nodes/vr_sros/vr-sros.go:func createVrSROSFiles(node nodes.Node) error {
```

That would be awesome it you could correct this separately. (thanks !)